src/main/java/io/airlift/airline/TypeConverter.java

```diff
@@ -54,7 +55,11 @@ else if (Double.class.isAssignableFrom(type) || Double.TYPE.isAssignableFrom(typ
                 return valueOf.invoke(null, value);
             }
         }
-        catch (Throwable ignored) {
+        catch (InvocationTargetException ex) {
+            throw new ParseOptionConversionException(name, value, type.getSimpleName(), ex.getTargetException());
```

Comment:
Instead of immediately throwing, save off the exception for the end, and try the other techniques.  The existing code tries all techniques, and I think we want to maintain that behavior.

Reply:
I thought about that, however, `InvocationTargetException` is only throw if the target, i.e. `fromString` or `valueOf` or the constructor was found, invoked and has failed. Ignore and try another method is only advisable if the method was not found or could not be invoked.

Reply:
The problem is this code just guesses which methods to call.  It is possible that the class had different parsing techniques for each of these methods, and we don't know which one is "correct".  This is why we try all and only fail if nothing works.

Reply:
@dain please see https://github.com/airlift/airlift/pull/759/commits/d6e67b8a794e622ec80dc02734a5caad26c159b0